src/aitsi/m3spin/commons/impl/TNodeImpl.java

```diff
@@ -0,0 +1,25 @@
+package aitsi.m3spin.commons.impl;
+
+import aitsi.m3spin.commons.enums.EntityType;
+import aitsi.m3spin.commons.interfaces.TNode;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@Getter
+@Setter
+@NoArgsConstructor
+@EqualsAndHashCode
+public abstract class TNodeImpl implements TNode {
+    TNode leftSibling;
+    TNode rightSibling;
+    TNode parent;
+    TNode firstChild;
+    TNode secondChild;
+    TNode thirdChild;
+    String attribute;
+    int stmtLine;
```

Comment:
Jeśli `stmtLine` ma oznaczać numer linii, w której znajduje się dana instrukcja (_statement_), to czy to pole nie powinno trafić do abstrakcyjnej klasy `StatementImpl`, którą rozszerzałyby implementacje `Assignment`, `While`, `If` oraz `Call`?

Reply:
Zastanawiam się tylko na razie